### ISSUE TITLE
Implement the transpose

### ISSUE THREAD

**Issue by lu-zero** [lu-zero/libvpx#8]

**Comment by shawnpringle** [comment]
Hello @lu-zero.  I have created a pull request for this issue https://github.com/lu-zero/libvpx/pull/14 .    You can use simple_decode.c to decode and transopose a video to raw frames.

I found that the vpx images need more memory than they are allocated for at times for some formats.  So the test program avoids vpx_img_alloc.

**Comment by lu-zero** [comment]
I'm afraid you misunderstood the whole set of issues. This list of issues (that is mainly my todo, so does not have much information) is about VSX and libvpx.

**Comment by shawnpringle** [comment]
okay.  so let's start over.  I think these are therequirements now: Routines Must use VSX instructions The routines must work in place but they can be for fixed sized square planes.  For what sized planes must they be and how many routines?

**Comment by shawnpringle** [comment]
There is a macro in  vpx_dsp/x86/inv_txfm_ssse3_x86_64.asm 
Are you looking for a VSX version of this?

```
    %macro TRANSPOSE8X8 9
      INTERLEAVE_2X  wd, %1, %2, %9
      INTERLEAVE_2X  wd, %3, %4, %9
      INTERLEAVE_2X  wd, %5, %6, %9
      INTERLEAVE_2X  wd, %7, %8, %9

      INTERLEAVE_2X  dq, %1, %3, %9
      INTERLEAVE_2X  dq, %2, %4, %9
      INTERLEAVE_2X  dq, %5, %7, %9
      INTERLEAVE_2X  dq, %6, %8, %9

      INTERLEAVE_2X  qdq, %1, %5, %9
      INTERLEAVE_2X  qdq, %3, %7, %9
      INTERLEAVE_2X  qdq, %2, %6, %9
      INTERLEAVE_2X  qdq, %4, %8, %9

      SWAP  %2, %5
      SWAP  %4, %7
    %endmacro
```

**Comment by shawnpringle** [comment]
Is this what is wanted?  Anyone?  Transpose16x16 macro that work for 8 bitdepth images and a transpose8x8 macro that is for high depth images.  Both implemented in Altivec instructions.  Please respond.

**Comment by lu-zero** [comment]
I hate repeating myself: this issue is not what you have in mind, you cannot start working on it without working on the other issues before.

Your behavior in the pull request managed to upset even @edelsohn and I really do not have time to teach you what is all about.

**Comment by shawnl** [comment]
This was fixed in dc93b6298b77a8e68c80175f2f7f5fb211be358a by Luc Trudeau <luc@trud.ca>
 @luctrudeau He should get the bounty.

**Comment by rafaeldelucena** [comment]
Actually the first transpose implementation for 8x8 matrix was added 405b94c committed at Oct 19, 2016.

But I do not know if only implementing the transpose for 8x8 signed int16 is enough to close the issue.

**Comment by shawnpringle** [comment]
The problem has always been a lack of spec.

As I recall, I also had proposed a pull-request for this one that was shot-down.  It was a C implementation.  I then found many other projects where similar situations.  Often the dev would implement the feature only to get several on the ticket saying that there ought to be a setting for this new feature to turn it on or off, or that the equal signs shouldn’t be aligned in the code.  It seemed to be always something designed to delay resolution indefinitely.

I now do things by negotiating and then doing coding contracts on closed source software outside of the bounty source site.  I now demand escrow agents hold the money rather than someone involved in the project.  And if the money gets released when the project leader closes the issue, this is essentially making someone involved in the project as an escrow agent’s controller.  If you say implement the transpose, any implementation of the transpose should be acceptable and the bounty should be paid.  The ambiguity is the fault of the one who creates the ticket.  Coders should not be expected to also be mind readers as well.  

I am sorry others have wasted time on this.

Sincerely,

Shawn Pringle

**Comment by shawnl** [comment]
@shawnpringle If they don't use your code, they don't owe you the bounty. Until that point you have not licensed the code under the project's license and have exclusive monopoly over the copying thereof. It is very simple, and would not work any other way. You wrote that code 2 years ago, and certainly have had time to try again, and with more respect.

**Comment by shawnpringle** [comment]
I agree with everything you just said.  No bitterness here.  I learned some from the experiencer.

Cheers,

Shawn Pringle
